system/include/emscripten/fetch.h

```diff
@@ -185,6 +186,22 @@ EMSCRIPTEN_RESULT emscripten_fetch_wait(emscripten_fetch_t *fetch, double timeou
 // onerror() handler will be called in the calling thread before this function returns.
 EMSCRIPTEN_RESULT emscripten_fetch_close(emscripten_fetch_t *fetch);
 
+// Gets the size (in bytes) of the response headers as plain text.
+// Note that this will return 0 if readyState < HEADERS_RECEIVED.
+size_t emscripten_fetch_get_response_headers_length(emscripten_fetch_t *fetch);
+
+// Gets the response headers as plain text. dstSizeBytes should be headers_length + 1 (for the null terminator).
+size_t emscripten_fetch_get_response_headers(emscripten_fetch_t *fetch, char *dst, size_t dstSizeBytes);
```

Comment:
There is an important limitation in play here with respect to these functions `emscripten_fetch_get_response_headers())` and `emscripten_fetch_get_response_headers_length()` that they must be called on the <b>same thread</b> where the fetch was initiated with `emscripten_fetch()`. This is because the `Fetch.xhrs` object is thread local. Can you add a comment to these two functions to note about this fact - `emscripten_fetch_get_response_headers_length()` and `emscripten_fetch_get_response_headers()` must be called on the same thread that originated the fetch?

Reply:
Done